src/app/services/calendar.service.ts

```diff
@@ -80,4 +82,48 @@ export default class CalendarService {
   getActiveDayIsOpen(): boolean {
     return this.activeDayIsOpen;
   }
-}
\ No newline at end of file
+
+  editEvent(event: CalEvent) {
+    this.http.put(
+      Globals.BACKEND + 'calendar/event/' + event.key,
+      this.mapCalendarEvent(event),
+      {
+        headers: this.authService.getTokenHeader()
+      }
+    ).subscribe(
+      response => {
+        this.events = this.events.map(evt => evt.key === event.key ? event : evt);
+        this.eventsChange.next(this.events);
+      },
+      err => console.log(err)
+    )
+  }
+
+  deleteEvent(event: CalEvent) {
+    this.http.delete(
+      Globals.BACKEND + 'calendar/event/' + event.key,
+      {
+        headers: this.authService.getTokenHeader(),
+        body: this.mapCalendarEvent(event)
+      }
+    ).subscribe(
+      response => {
+        this.events = this.events.filter(evt => evt.key !== event.key);
+        this.eventsChange.next(this.events);
+      },
+      err => console.log(err)
+    )
+  }
+
+  private mapCalendarEvent(event: CalEvent): any {
```

Comment:
So this function can now be used up in `mapCalendarEvents()`, no? (lines ~39-45)